WalletWasabi/Hwi/ProcessBridge/HwiProcessBridge.cs

```diff
@@ -1,42 +1,43 @@
 using System;
-using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
-using System.Runtime.InteropServices;
-using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
-using WalletWasabi.Helpers;
-using WalletWasabi.Interfaces;
-using WalletWasabi.Logging;
 using WalletWasabi.Microservices;
 
 namespace WalletWasabi.Hwi.ProcessBridge
 {
-	public class HwiProcessBridge : Microservices.ProcessBridge
+	public class HwiProcessBridge : IHwiProcessInvoker
 	{
-		public HwiProcessBridge() : base(MicroserviceHelpers.GetBinaryPath("hwi"))
+		public HwiProcessBridge(ProcessInvoker processInvoker)
```

Comment:
Inheriting from `IHwiProcessInvoker` and holding a `ProcessInvoker`. This just doesn't seem right.

Reply:
* `IHwiProcessInvoker` is an interface and it is used to allow mocking.
* `ProcessInvoker processInvoker` is used to access features of that class. I tend to avoid extending classes and I use rather class composability. It allows to build functionality from bottom-up.

Reply:
Quoting you:

> The ProcessBridge API is more complex than necessary

It seems like you aren't really achieving your goal. Making it more complex than it was before.